packages/graph-engine/src/nodes/generic/group.ts

```diff
@@ -0,0 +1,8 @@
+import { Node } from '../../programmatic/node.js';
+
+export default class NodeDefinition extends Node {
```

Comment:
I don't think we should add all this. This node does nothing, we've cluttered a lot of the code and we're likely going to get inconsistencies from the annotations not being updated correctly between reactflow. We should rather just use the finalizers for serialization and deserialization and store the group info as annotations until we make a more concrete construct that actually does stuff.